packages/block-library/src/template-part/edit/index.js

```diff
@@ -100,10 +104,15 @@ export default function TemplatePartEdit( {
 						/>
 					</div>
 				</BlockControls>
-				<TemplatePartInnerBlocks
-					postId={ postId }
-					hasInnerBlocks={ innerBlocks.length > 0 }
-				/>
+				<div
+					onMouseEnter={ () => addHoveredEntity( clientId ) }
+					onMouseLeave={ () => removeHoveredEntity( clientId ) }
+				>
```

Comment:
This didn't work on the `BlockWrapper`, the `onMouseEnter` worked there, but `onMouseLeave` was getting overridden.

Adding this ad-hoc doesn't seem bad given we don't have many entities we need to highlight in the document settings dropdown.  However, if we did wan't this functionality to be more reusable maybe we could add these handlers via a supports hook similar to the style hooks? 🤔